### PR TITLE
[Cleanup] Deduplicate kernel tests, diffusion fixtures and benchmark helpers

### PR DESCRIPTION
**Pull request by BBuf**
## Motivation

Kernel and diffusion tests repeat input/reference code, fixture setup and whole test bodies. Some benchmarks also carry copied or unused helpers and stale import paths. This cleanup removes 1,426 net lines across 26 files while retaining the numerical checks, backend-specific dispatch, input matrices and CI registrations.

## Modifications

- Share paged-MQA fixtures, PyTorch reference and comparison logic between CuTe DSL and DeepGEMM tests; retain both complete backend matrices. Share the original AWQ input packer between dense and MoE repack tests.
- Parameterize metadata-copy small/large batches into one body per backend. Keep all 40 numerical cases and both dtype rejection tests. Delete unreachable reference branches and single-shot performance-print scaffolding from correctness tests.
- Consolidate diffusion gate/capture/fallback cases, FP32 LayerNorm cache checks and VAE defaults. Fold upsampler call counts into numerical tests and replace the copied full-model reference with an unfused GroupNorm + activation override; keep the independent ResBlock oracle and original CPU/GPU case boundaries. Reuse the existing parent server-args fixture in three SANA-WM suites.
- Share DeepSeek-V3 benchmark shape tables and FP8 input preparation; fix grouped-GEMM helper imports and use DeepGEMM's current testing-module import for `calc_diff`. Preserve the original unrounded FP32 scales rather than substituting the production UE8M0 quantizer.
- Delete the copied DeepEP benchmark utility file and reuse existing test helpers; keep its CLI-specific distributed initializer unchanged. Remove unused FA4 backward/profiling helpers, an uncalled vLLM GEMM wrapper/import and three uncalled MMMU data helpers.

## Accuracy Tests

Rebased onto main `d507accadc` and validated on GB300 (SM103), PyTorch 2.13.0+cu130:

| Existing tests | Result |
| --- | --- |
| CuTe DSL + DeepGEMM paged MQA, full matrices | 384 passed |
| Diffusion fusion, norm/cache, fallback and VAE loader | 69 passed, 8 subtests passed |
| Residual gate | 35 passed |
| Metadata copy + AWQ repack | 58 passed, 30 skipped |
| SANA-WM streaming/realtime suites | 23 passed |

Total: **569 passed, 30 skipped, 8 subtests passed**. The 30 skips are JIT-vs-AOT comparisons because this environment does not expose the old AOT repack operators; the independent JIT numerical checks ran. Metadata/AWQ and SANA-WM results match the before-cleanup run on the same base. Original first-pass diffusion cases also passed before consolidation. No numerical tolerance was relaxed.

Additional validation:

- 88 old/new FP8 payload-and-scale comparisons (CPU/CUDA, BF16/FP32, zero and padded inputs), 24 exact AWQ packer comparisons and all four original shape tables at TP 1/2/4/8/16 match.
- The generated metadata-copy parameter matrix is identical; all 11 CI registration calls are unchanged. Retained MMMU functions are AST-identical, with four prompt-construction smoke checks.
- All four affected DeepSeek/block-wise benchmark CLIs pass `--help` import smoke checks. Shared MQA and moved/reused DeepEP helpers retain their original implementations. All applicable pre-commit hooks passed.

## Speed Tests and Profiling

No inference behavior changes or inference speedup claim. The only runtime-file edit is a formatting-only change in `logits_processor.py` required by the repository-wide Ruff check (AST unchanged). DeepEP helpers pass 18 exact tensor/scalar comparisons and CUDA-event timer smoke checks; the retained FA4 forward timer passes with AMP off/on. Benchmark providers and timed callbacks are unchanged. Full multi-node DeepEP and GEMM tuning sweeps were not run.

## Checklist

- [x] Format code with pre-commit.
- [x] Run the affected existing tests; no new test suite added.
- [x] Review numerical coverage, backend guards and CI registrations.
- [x] Document validation and its scope above.

---
### CI States

Latest PR Test (Base): :x: [Run #35437052621](https://github.com/sgl-project/sglang/actions/runs/35437052621)
Latest PR Test (Extra): :x: [Run #35437052436](https://github.com/sgl-project/sglang/actions/runs/35437052436)
Latest PR Test (AMD ROCm 10): :hourglass_flowing_sand: [Run #35437052589](https://github.com/sgl-project/sglang/actions/runs/35437052589)

### CODE REVIEW

**Review by mickqian — COMMENTED** [review]
CI follow-up for head c28ff8615a57c9b00e0af55ac9bcbfe3fc2a384c: Lint job 105812171744 fails because pre-commit reformats python/sglang/srt/layers/logits_processor.py around line 467, collapsing the parallel.attn_tp_group / self.use_attn_tp_group / parallel.tp_group conditional onto one line inside the existing parentheses. This file is not part of this PR diff, so this is not evidence of a regression introduced by this cleanup. Please apply the exact formatter change (or sync the target if already fixed), then rerun the relevant pre-commit hook. I do not have push permission to BBuf/sglang; no branch changes were made. Evidence: https://github.com/sgl-project/sglang/actions/runs/35411603749/job/105812171744

**Comment by BBuf** [comment]
/rerun-test python/sglang/multimodal_gen/test/unit/sana_wm/test_realtime_chain.py python/sglang/multimodal_gen/test/unit/sana_wm/test_streaming_forward_long.py python/sglang/multimodal_gen/test/unit/sana_wm/test_streaming_stage.py python/sglang/multimodal_gen/test/unit/test_fp32_layernorm.py python/sglang/multimodal_gen/test/unit/test_latent_upsampler_group_norm_silu.py python/sglang/multimodal_gen/test/unit/test_layernorm_cutedsl_dispatch.py python/sglang/multimodal_gen/test/unit/test_vae_loader.py test/registered/kernels/ops/attention/test_cute_dsl_fp8_paged_mqa_logits.py test/registered/kernels/ops/attention/test_deepgemm_paged_mqa_logits.py test/registered/kernels/ops/attention/test_fused_metadata_copy.py test/registered/kernels/ops/diffusion/test_flux2_token_cat_fp8.py test/registered/kernels/ops/diffusion/test_modulate.py test/registered/kernels/ops/quantization/test_awq_marlin_moe_repack.py test/registered/kernels/ops/quantization/test_awq_marlin_repack.py

**Comment by BBuf** [comment]
/rerun-test python/sglang/multimodal_gen/test/unit/sana_wm/test_realtime_chain.py python/sglang/multimodal_gen/test/unit/sana_wm/test_streaming_forward_long.py python/sglang/multimodal_gen/test/unit/sana_wm/test_streaming_stage.py python/sglang/multimodal_gen/test/unit/test_fp32_layernorm.py python/sglang/multimodal_gen/test/unit/test_latent_upsampler_group_norm_silu.py python/sglang/multimodal_gen/test/unit/test_layernorm_cutedsl_dispatch.py python/sglang/multimodal_gen/test/unit/test_vae_loader.py test/registered/kernels/ops/attention/test_cute_dsl_fp8_paged_mqa_logits.py test/registered/kernels/ops/attention/test_deepgemm_paged_mqa_logits.py test/registered/kernels/ops/attention/test_fused_metadata_copy.py test/registered/kernels/ops/diffusion/test_flux2_token_cat_fp8.py test/registered/kernels/ops/diffusion/test_modulate.py test/registered/kernels/ops/quantization/test_awq_marlin_moe_repack.py test/registered/kernels/ops/quantization/test_awq_marlin_repack.py test/registered/unit/managers/test_disagg_idle_step_counters.py
